sourcegraph/sourcegraph
# Add exhaustive search creation UI

Closes https://github.com/sourcegraph/sourcegraph/issues/55707
Partially closes https://github.com/sourcegraph/sourcegraph/issues/55973

[Figma desings](https://www.figma.com/file/H4ezuXmEvVeoC6UaiNSbEa/Exhaustive-search-UI?type=design&node-id=1568-10052&mode=design&t=3su0QK4wsCCTC082-0)

| New UI with no problems in query | With validation problems |
| -----  | ----- | 
| <img width="737" alt="Screenshot 2023-08-25 at 00 57 36" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/d4f28ed7-c928-48e3-94a3-ea53bc60ad5e"> | <img width="729" alt="Screenshot 2023-08-25 at 00 57 51" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/710cf1e5-09d4-4f1b-9bfa-31b3ad7184f0"> |

## Tech notes 
In this PR we add an initial creation form UI in skipped results popover and also add some query validation algoright (https://github.com/sourcegraph/sourcegraph/issues/55973), but this validation isn't comprehensive since we probably want to check the query in a more sophisticated way (like analyse OR operand and check filter in operands like @stefanhengl suggested in this [comment](https://github.com/sourcegraph/sourcegraph/issues/55973#issuecomment-1688429040) and not just check operator existence in the query) 

## Design review notes 
- The experimental badge has a different colour/font capital compared to what we have in design, @rrhyne let me know if it's a misalignment in wildcard implementation, our storybook [demo](https://storybook.sgdev.org/?path=/story/wildcard-productstatusbadge--badges)
-  In Figma, we have just block with an alert warning border but not an Alert warning border. I used the Alert component here since we have [them in the wildcard](https://storybook.sgdev.org/?path=/story/wildcard-alert--alerts)

## Test plan
- Turn on `exhaustiveSearch:true` in your `experimentalFeatures` in your settings editor
- Go to search and try to search for something that may produce more results that we can show on the search result page 
- Check the skipped results button. There should be an exhaustive search (search job) creation UI
